backend/server.py

```diff
@@ -110,152 +178,182 @@ class Measurement(db.Model):
     @classmethod
     def create(cls, measurement_data, hierarchy_id, node_id):
         new_measurement=Measurement(
-            node_id=node_id,
             hierarchy_id=hierarchy_id,
+            node_id=node_id,
 
             name=measurement_data["measurementName"],
             type=measurement_data["measurementType"],
-
-                
         )
 
         db.session.add(new_measurement)
         db.session.commit() # measurement now has a unique id
 
-        return new_measurement.id
+        return new_measurement
 
     @classmethod
     def create_measurements(cls, measurements_lst, hierarchy_id, node_id):
         for measurement_data in measurements_lst:
             Measurement.create(measurement_data, hierarchy_id, node_id)
 
+    @classmethod
+    def get(cls, measurement_id):
+        return Measurement.query.filter_by(id=measurement_id).first()    
+
+    @classmethod
+    def get_all(cls, hierarchy_id, node_id):
+        return Measurement.query.filter_by(hierarchy_id=hierarchy_id, node_id=node_id)
+
+    def to_dict(self):
+        measurement_dict = {
+                "id": str(self.id),
+
+                "name": self.name,
+                "type": self.type,
+                "valueFunction": self.value_function,
+            }
+        
+        return measurement_dict
+
+    @classmethod
+    def get_list(cls, hierarchy_id, node_id):
+        measurements = Measurement.get_all(hierarchy_id, node_id)
+        measurement_list = []
+
+        for measurement in measurements:
+            measurement_list.append(measurement.to_dict())
+        
+        return measurement_list
+
 
 # routes
 @app.route("/", methods=['GET'])
 def hello_world():
     return "<p>Hello World!</p>"
 
 
+# TODO Add checks that hierarchies/nodes exist before adding subordinate resources
 @app.route("/hierarchy", methods=['POST'])
 def create_hierarchy():
     # Get data from .json sent with request
     data = request.get_json()
 
     # Create Hierarchy
-    new_hierarchy_id = Hierarchy.create(data)
+    new_hierarchy = Hierarchy.create(data)
 
     # Parse and create Nodes
     nodes_lst = data["nodes"]
-    Node.create_nodes(nodes_lst, new_hierarchy_id)
+    Node.create_nodes(nodes_lst, new_hierarchy.id)
 
-    return jsonify({"message": "Hierarchy Created"}, 200)
+    # Create a hierarchy dict with all needed data
+    hierarchy_dict = Hierarchy.get_dict(new_hierarchy.id)
 
+    return jsonify(201, hierarchy_dict)
 
-@app.route("/hierarchy/ascending_id", methods=['GET'])
-def get_all_hierarchies_ascending():
-    hierarchies = Hierarchy.query.all()
-    all_hierarchies = []
 
-    for hierarchy in hierarchies:
-        all_hierarchies.append({
-            "id": hierarchy.id,
-            "name": hierarchy.name,
-            "description": hierarchy.description,
-        })
+@app.route("/hierarchy/<hierarchy_id>/node/<parent_id>", methods=['POST'])
```

Comment:
What if you want to add a node to the root of the hierarchy?

Reply:
For now you'd send a parent_id of 0.

There may be a better way to model the database by using the root node as a replacement for the current Hierarchy Model. It would require a refactor, but it may be simpler.